scipy/optimize/optimize.py

```diff
@@ -483,7 +482,8 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
 
     iterations = 1
 
-    while (fcalls[0] < maxfun and iterations < maxiter):
+    while ((maxfun is None or fcalls[0] < maxfun)
+        and (maxiter is None or iterations < maxiter)):
```

Comment:
Yes, that would be a bit more readable.

So default them both to np.inf in the function definition. And then the user does not set either of them set them to N*200. Or should the N*200 be dropped altogether (personally it seems sensible to have a max iterations in the default case).